rust/arrow/src/record_batch.rs

```diff
@@ -217,15 +217,12 @@ impl Into<StructArray> for RecordBatch {
 }
 
 /// Trait for types that can read `RecordBatch`'s.
-pub trait RecordBatchReader {
+pub trait RecordBatchReader: Iterator<Item = Result<RecordBatch>> {
     /// Returns the schema of this `RecordBatchReader`.
     ///
     /// Implementation of this trait should guarantee that all `RecordBatch`'s returned by this
     /// reader should have the same schema as returned from this method.
     fn schema(&self) -> SchemaRef;
-
-    /// Reads the next `RecordBatch`.
```

Comment:
I wonder if you could leave this function in the trait (and provide a default implementation it in terms of `self.next()`) if you wanted to reduce the changes required for existing code (maybe it could be marked as deprecated?)

Reply:
Done. Really good idea, @alamb. Thanks a lot for this and all the other reviews! (fyi, I can't merge to master yet, so we both need to wait for someone to take these and yours 🕐)